packages/mwp-tracking-plugin/src/util/avro.js

```diff
@@ -42,17 +41,40 @@ const getPlatformAnalyticsLog = (
 const getLogAWSKinesis = (usePubSub: ?string = canUsePubSub): (string => void) => {
 	if (usePubSub) {
 		return (serializedRecord: string) => {
-			const options = {
-				Data: Buffer.from(serializedRecord),
-				PartitionKey: uuidv1(),
-				StreamName: process.env.KINESIS_STREAM_NAME,
-			};
+			const AWS = require('aws-sdk');
+			const sts = new AWS.STS();
+			sts.assumeRole(
+				{
+					RoleArn: process.env.ASSUME_ROLE_ARN,
+					RoleSessionName: 'awssdk',
+				},
+				function(err, data) {
+					if (err) {
+						// an error occurred
+						console.log('Cannot assume role');
+						console.log(err, err.stack);
```

Comment:
print stacktrace is not very cloudwatch friendly, it would spread into multiple lines.

Reply:
I will update once it proved work.